src/LogFile.cpp

```diff
@@ -98,7 +98,7 @@ LogFormat(const char *fmt, ...) noexcept
   va_list ap;
 
   va_start(ap, fmt);
-  vsprintf(buf, fmt, ap);
+  vsnprintf(buf, sizeof(buf) - 1, fmt, ap);
```

Comment:
Off-by-one bug. Why `- 1`?

Reply:
Because the ending 0, but this is vsnprintf doing by itself, I think!

Reply:
> The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str.